instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go

```diff
@@ -84,12 +85,11 @@ func UnaryClientInterceptor(opts ...Option) grpc.UnaryClientInterceptor {
 		}
 
 		name, attr := spanInfo(method, cc.Target())
-		var span trace.Span
-		ctx, span = tracer.Start(
+		startOpts := append(defaultStartOpts, trace.WithAttributes(attr...))
```

Comment:
This is not safe to do as this function can be called concurrently from multiple goroutines and the backing array of the `defaultStartOpts` might be modified concurrently by this `append`. This can only happen if the slice happens to have unused capacity i.e. the array is re-used rather than re-allocated, but there is a chance nonetheless. I hope this makes sense.

A solution is to explicitly allocate a new slice, copy elements from `defaultStartOpts` and append `trace.WithAttributes()`.

Reply:
You are correct, I didn't notice that when moving the slice creation to outside of the inner function, and that can definitely happen. My idea was to avoid calling `trace.WithSpanKind` and `cfg.SpanStartOptions...` for every span since those options never change. Do you think thats a worthwhile optmization?
If not, I can always revert to creating the whole slice for each span which would avoid this problem.

Reply:
It probably is a minor but worthwhile optimization. The code will be allocating a slice one way or another so no harm in making it a bit more efficient even if a bit more clunky.

Reply:
On the other hand, maybe Go compiler can optimize/inline the allocation/call of the option function and there is no point. I'll leave this to maintainers to decide.

Reply:
Sorry for leaving this unsolved, I chose to revert the changes to avoid the problem.